cookbooks/aws-parallelcluster-config/spec/unit/resources/manage_raid_mount_unmount_spec.rb

```diff
@@ -0,0 +1,196 @@
+
+        command = chef_run.execute("create_raid_config")
+        expect(command).to subscribe_to('execute[setup_raid_disk]').on(:run).immediately
+      end
+
+      it "Create the shared dir" do
+        is_expected.to create_directory('/raid_shared_dir')
+          .with(owner: 'root')
+          .with(group: 'root')
+          .with(mode: '1777')
+        # recursive do not work
+        # .with(recursive: true)
+      end
+
+      it "Mount and enable the shared dir" do
+        is_expected.to mount_mount('/raid_shared_dir')
+          .with(device: "/dev/md0")
+          .with(fstype: "ext4")
+          .with(options: %w(defaults nofail _netdev))
+          .with(retries: 10)
+          .with(retry_delay: 6)
+
+        is_expected.to enable_mount('/raid_shared_dir')
+          .with(device: "/dev/md0")
+          .with(fstype: "ext4")
+          .with(options: %w(defaults nofail _netdev))
+          .with(retries: 10)
+          .with(retry_delay: 6)
+      end
+
+      it "Change permission for the shared dir" do
+        is_expected.to create_directory('/raid_shared_dir')
+          .with(owner: 'root')
+          .with(group: 'root')
+          .with(mode: '1777')
+      end
+    end
+  end
+end
+
+describe 'manage_raid:unmount' do
+  for_all_oses do |platform, version|
+    context "on #{platform}#{version}" do
+      cached(:venv_path) { 'venv' }
+      cached(:chef_run) do
+        runner = ChefSpec::Runner.new(
+          platform: platform, version: version,
+          step_into: ['manage_raid']
+        ) do |node|
+          node.override['cluster']['cookbook_virtualenv_path'] = venv_path
+        end
+        runner.converge_dsl do
+          manage_raid 'unmount' do
+            action :unmount
+            raid_vol_array "vol-0, vol-1"
+            raid_shared_dir "raid_shared_dir"
+            raid_type ' 0'
+          end
+        end
+      end
+
+      it "Unmount and disable shared dir" do
+        is_expected.to unmount_mount("/raid_shared_dir")
+          .with(device: "/dev/md0")
+          .with(fstype: "ext4")
+          .with(options: %w(defaults nofail _netdev))
+          .with(retries: 10)
+          .with(retry_delay: 6)
+
+        is_expected.to disable_mount("/raid_shared_dir")
+          .with(device: "/dev/md0")
+          .with(fstype: "ext4")
+          .with(options: %w(defaults nofail _netdev))
+          .with(retries: 10)
+          .with(retry_delay: 6)
+      end
+
+      it "Delete shared dir" do
+        is_expected.to delete_directory("/raid_shared_dir").with(recursive: true)
+      end
+
+      it "Stop RAID device" do
+        is_expected.to run_execute("Deactivate array, releasing all resources")
+          .with(command: "mdadm --stop /dev/md0")
+      end
+
+      stubs_for_provider('manage_raid') do |resource|
+        allow(resource).to receive(:get_raid_devices).with("/dev/md0").and_return("device1 device2")
+      end
```

Comment:
Nitpick. I would move it before examples, as the first instruction of this context, because this is part of the context setup.